### PR TITLE
fix when fetching tag and unspecified branch

### PR DESCRIPTION
**Pull request by teto**
follow up of https://github.com/nix-community/poetry2nix/pull/167 . Explanation at https://github.com/nix-community/poetry2nix/issues/197 . I would like to update the tests but not sure what's best.

### CODE REVIEW

**Comment by teto** [comment]
It doesn't seem it's possible in poetry to specify both a rev and a ref ? for htis reason I can't fetch some specific revisions that are not on the default branch.

File: mk-poetry-dep.nix [hunk 1]
```diff
@@ -152,14 +152,18 @@ pythonPackages.callPackage
...
       # Interpreters should declare what wheel types they're compatible with (python type + ABI)
       # Here we can then choose a file based on that info.
-      src =
+      src = let
+          # tagRef = if (sourceSpec ? tag) then "refs/tags/${sourceSpec.tag}" else null;
+        in
         if isGit then
           (
-            builtins.fetchGit {
+            builtins.fetchGit (lib.debug.traceValSeq ({
```
Comment (Mic92) [inline mk-poetry-dep.nix:160]: This should be not left it no, no?

Comment (teto) [reply mk-poetry-dep.nix:160]: yes it should go away :)  I am not sure this whole PR is a definitive fix nor how to operate/update the tests so I dont intend to complete this as of now. Anyone feel free to pick it up.

**Comment by teto** [comment]
I think I have my final solution, if and only if there is a branch or tag add { ref = REF } (with REF=refs/tags/TAG if a tag) and nothing otherwise (I tried to pass ref=null instead of ref="HEAD": but nix doesnt like it).
I would like to see a fix merged but I don't trust much the test infra at the moment, should I ?

**Comment by adisbladis** [comment]
> I think I have my final solution, if and only if there is a branch or tag add { ref = REF } (with REF=refs/tags/TAG if a tag) and nothing otherwise (I tried to pass ref=null instead of ref="HEAD": but nix doesnt like it).

This has now been merged on master in https://github.com/nix-community/poetry2nix/pull/240

**Pull request closed without merging**